enterprise/core-edge/src/main/java/org/neo4j/coreedge/raft/log/segmented/Terms.java

```diff
@@ -107,7 +107,30 @@ synchronized void truncate( long fromIndex )
     synchronized void prune( long upToIndex )
     {
         min = max( upToIndex, min );
-        // could also prune out array
+
+        int offset = find( min );
+
+        if ( offset == -1 )
+        {
+            return;
+        }
+
+        size = 1;
```

Comment:
This is the size of the array, you cannot reset it to 1. You should set it to the size of the remainder of the array.